### DISCUSSION TITLE
HyperSPI on ESP8266 running from Rpi4

### DISCUSSION THREAD

**Discussion by houseofquad1974** [awawa-dev/HyperSPI#10] [Q&A]
Hi, I hope this is ok to start a new thread. If it's a 'nuisance', let me know and I will remove it.

After talking with @awawa-dev in another thread, i thought i would make a specific post in regards to a setup that hasn't been discussed yet. It seems no two setups are ever identical. Also, not that i can offer anything in the way of technical advice, i simply thought this added info could be of use, even if only for one other person.

Setup:
Raspberry Pi4 (dedicated 5v USB power supply)
SK6812RGBW 60LED's/m (Cool/Cold White) LED Strip (232 LED total / 15.3A max)
Cheap generic USB grabber. (DigitNow USB3.0 HDMI Video Capture)
Generic Power Supply - 230V / 5V 20A.
AZ Delivery ESP8266 NodeMCU V3. (advertised WITH CH340 chip)

To add to setup:
Level shifter (if needed)

As discussed in another thread, i had attempted to try the faster, recommended setup using an ESP8266 or ESP32 method. I tried with ESP8266 and failed to get the lights working. Anyway, i think i may have had 2 connections (GPIO 12 and 14 [green and blue]) the wrong way round. It's dark in behind my TV and so my old eyes probably mixed them up when joining the pairs of jumper leads between the module and the pi. My hope is that that is all that was wrong with my initial attempt at this.

Edit: Second attempt was successful, so it does appear that i made an error connecting up the first time.

Anyway, another reason for this thread is that i can post a picture of my module and confirm i am connecting up correctly. Just for clarity, i AM including a COMMON GROUND between pi and module. (see second image).

As always, HUGE thanks to @awawa-dev and his undying devotion to this project.

![IMG_3770](https://user-images.githubusercontent.com/60320875/206864596-96256f56-1f3f-42d0-8629-2673384a47ce.jpg)

![IMG_3784](https://user-images.githubusercontent.com/60320875/206922331-04b97d08-cc3a-4b73-8362-3e4265b0aea1.jpg)

**Comment by awawa-dev** [comment]
Hi
Discussions here are the perfect place for such topics. Unfortunately, we don't see a connection for the Rpi pinout, so I trust you to connect it correctly. I will point out a few things that may help narrow down or even solve the problem.
1) The length of cable I used in the project is around 15cm, the longer cables the bigger risk for spi communication (and we use very high speed)
2) I know you connected the ground, but it's absent on the picture so I want just to make sure for other users to connect ground from Rpi pinout to Esp pinout also.
3) You have Esp8266 so in HyperHDR AWA spi driver properties you must select esp8266 option
4) Since you have ESP connected to the Rpi try to use serial client to read some statistics from HyperSPI
a) install `screen` (sudo apt install screen)
b) run screen e.g. `screen /dev/ttyUSB0 115200` adjust USB port if necessary  (if you want to exit screen press Ctrl-a then k to exit screen later)
c) try to disable and enable the HyperSPI device in HyperHDR, watch the output from `screen` (wait 5-6 second for the stats appear in the `screen` after switching LEDs on/off)
4) if everything else fails, lower the SPI speed to 10 000 000

**Reply by houseofquad1974** [reply]
It seems that i've encountered an 'issue' with the latest setup. Pretty sure it's going to be something i've missed/haven't done.

When I turn on HDR to SDR mapping, the lights all turn off and no longer respond to the screen. When i deselect the option, the lights all spring back into action. Any suggestions what as to what i may have missed ? Are any of the calibrations required before this option will work on v19 ? Previously (on v17) i could turn on this option and the LED strip would 'go up a few notches' to the point that it seemed 'over-cooked' ! I never did any calibrations though, so i understand why the lights weren't behaving as preferred. I never bothered to calibrate, purely because i was happy with the lights as they were. I still am now, even without any calibrations. I just thought i would make a note of this 'issue' in case it was something more than 'my fault'. Also, as things stand, and going by all the latest improvements, i am still probably not getting the best available effect out of this setup. Eventually i will delve deeper into the calibrations.

Would you @awawa-dev be interested in seeing a video of how they are at the moment ? I can compile a video for youtube of the original v17 setup, and then also one of this latest v19 setup.

As always, thanks for all your endeavours.

**Reply by awawa-dev** [reply]
Hi
> When I turn on HDR to SDR mapping, the lights all turn off and no longer respond to the screen. When i deselect the option, the lights all spring back into action. Any suggestions what as to what i may have missed ?

I think the most important thing is HyperHDR logs: enable HDR tone mapping and restart the Rpi, wait 2-3 minutes after it starts again and copy the logs to the pastebin.com providing us a link. Could create a new topic(you can paste a copy) in HyperHDR discussion as the issue is directly related to the application? Some things that are also worth to check out:
- if you have a custom LUT table in your /home/pi/.hyperhdr then delete it 
- if your grabber supports different encoding e.g. MJPEG, YUV, NV12 then try to use the different one
